### PR TITLE
Abililty to configure provider by region, not having to specify endpoints anymore

### PR DESCRIPTION
**Pull request by Soulou**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,12 +21,37 @@ Requirements
...
+
 provider "scalingo" {
-  api_token = "MY_API_TOKEN"
+  api_token = "${var.scalingo_api_token}"
+  region = "osc-fr1"
 }
 ```
 
+```bash
+export TF_VAR_scaling_api_token=tk-1234567890
```
Comment (EtienneM) [inline README.md]:
```suggestion
export TF_VAR_scaling_api_token=tk-us-1234567890
```

File: README.md [hunk 2]
```diff
@@ -21,12 +21,37 @@ Requirements
...
+
+### Configuration with environment variables
+
+```
+provider "scalingo" {}
+```
+
+```bash
+export SCALINGO_REGION=osc-fr1
+export SCALINGO_API_TOKEN=tk-1234567890
```
Comment (EtienneM) [inline README.md]:
```suggestion
export SCALINGO_API_TOKEN=tk-us-1234567890
```

File: scalingo/provider.go [hunk 3]
```diff
@@ -17,12 +19,22 @@ func Provider() terraform.ResourceProvider {
 			"api_url": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("SCALINGO_API_URL", "https://api.scalingo.com/"),
+				DefaultFunc: schema.EnvDefaultFunc("SCALINGO_API_URL", nil),
+			},
+			"db_api_url": {
```
Comment (EtienneM) [inline scalingo/provider.go]: Is it really needed to specify the DB API URL and the API URL? Just the region is enough, isn't it?

Comment (Soulou) [reply scalingo/provider.go]: For the production public environment no, but for dev/staging/private deployments, yes :)

**Pull request merged**